builtins/amp-img.js

```diff
@@ -89,6 +84,14 @@ export class AmpImg extends BaseElement {
     return promise;
   }
 
+  /** @override */
+  layoutCallback() {
+    // This is specifically needed for elements that does its work on prerender
+    // e.g. amp-img. This is only needed because currently prerendering
+    // undisplayed elements does not work. See issue #2742.
+    return this.updateImageSrc_();
```

Comment:
True. Good. This is starting to get us closer to the desired behavior. You are also right about use of two images - obviously a much better approach. A good guideline - keep everthing separate. This is also very similar to our current placeholder setup.